pkg/dynamicaccess/accesslogic.go

```diff
@@ -11,54 +11,65 @@ import (
 
 var hashFunc = sha3.NewLegacyKeccak256
 
-// Logic has the responsibility to return a ref for a given grantee and create new encrypted reference for a grantee
-type Logic interface {
+// Read-only interface for the ACT
+type Decryptor interface {
+	// DecryptRef will return a decrypted reference, for given encrypted reference and grantee
+	DecryptRef(rootHash swarm.Address, encryped_ref swarm.Address, publisher *ecdsa.PublicKey) (swarm.Address, error)
+	// Embedding the Session interface
+	Session
+}
+
+// Control interface for the ACT (does write operations)
+type Control interface {
+	// Embedding the Decryptor interface
+	Decryptor
 	// Adds a new grantee to the ACT
-	AddNewGranteeToContent(rootHash swarm.Address, publisherPubKey, granteePubKey *ecdsa.PublicKey) (swarm.Address, error)
-	// Get will return a decrypted reference, for given encrypted reference and grantee
-	Get(rootHash swarm.Address, encryped_ref swarm.Address, publisher *ecdsa.PublicKey) (swarm.Address, error)
+	AddGrantee(rootHash swarm.Address, publisherPubKey, granteePubKey *ecdsa.PublicKey, accessKey *encryption.Key) (swarm.Address, error)
+	// Encrypts a Swarm reference for a given grantee
+	EncryptRef(rootHash swarm.Address, grantee *ecdsa.PublicKey, ref swarm.Address) (swarm.Address, error)
 }
 
 type ActLogic struct {
-	session Session
-	act     Act
+	Session
```

Comment:
Does Session need to be exported ?

Reply:
I think Viktor told us to export it. We don't fully understand the reason for it. Probably it shouldn't be exported.